nuitka/utils/StaticLibraries.py

```diff
@@ -189,6 +191,23 @@ def _getSystemStaticLibPythonPath():
                 # about that anyway.
                 pass
 
+        if isPyenvPython():
+            try:
+                machine = subprocess.check_output(["gcc", "-dumpmachine"])
```

Comment:
I wonder if this lives somewhere in `sysconfig`. Usage or availability of GCC seems not a given. The value of `LIBPL`  could be that directory.

On my macOS it seems that way, and on Linux self compiled I got that:

```
>>> import sysconfig
>>> sysconfig.get_config_var("LIBPL")
'/opt/python3110/lib/python3.11/config-3.11-x86_64-linux-gnu'
```

And actually the static library lives there, so maybe it's not even specific.

Reply:
So, I am thinking, you can make this not depend on the flavor at all, and I think it will immediately optimize self-compiled Python on Linux properly.